chain/network/src/peer_manager/peer_manager_actor.rs

```diff
@@ -852,7 +853,26 @@ impl PeerManagerActor {
                     None
                 }
             })
-            .collect::<Vec<_>>()
+            .collect::<Vec<_>>();
+
+        // remove peers to which we have no route from the list
+        result.retain(|peer| {
+             if self
+                .routing_table_view
+                .peer_forwarding
+                .get(&peer.peer_info.id)
+                .unwrap_or(&Vec::new())
+                .is_empty()
+            {
+                error!(target: "network", "returning peer to ClientActor to which we have no route: {}", &peer.peer_info.id);
```

Comment:
how can we make sure that this doesn't spam our logs?

Reply:
Ok, I can add something that limits number of prints.